spec/src/main/java/net/neoforged/javadoctor/spec/ClassJavadoc.java

```diff
@@ -50,7 +50,7 @@ public Map<String, ClassJavadoc> innerClasses() {
     public ClassJavadoc merge(@Nullable ClassJavadoc other) {
         if (other == null) return this;
         return new ClassJavadoc(
-                clazz == null ? other.clazz : clazz,
+                clazz == null ? other.clazz : other.clazz == null ? clazz : clazz.merge(other.clazz),
```

Comment:
```suggestion
                clazz == null ? other.clazz : clazz.merge(other.clazz),
```
`JavadocEntry#merge` should be able to handle a null parameter.

Reply:
It wasn't, but I fixed that too